specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp

```diff
@@ -158,7 +158,7 @@ model ChatChoice {
 
     @doc("The reason that this chat completions choice completed its generated.")
     @projectedName("json", "finish_reason")
-    finishReason: CompletionsFinishReason;
+    finishReason: CompletionsFinishReason | null;
```

Comment:
I think a linter rule was recently added discouraging this `| null`. Is `finishReason?: CompletionsFinishReason` inaccurate?

Reply:
This intersects really well the point that @joseharriaga raised just a bit earlier around `logprobs`: what's the appropriate representation of optionality vs. nullability?

Here, specifically, I believe `finish_reason` *always* appears in a `Choice`, but in one specific situation (in-progress streaming responses), it supports having a `"finish_reason": null` assignment (presumably representing "there's no finish reason because I'm not finished yet).

Theoretically, we could have "required and non-nullable," "required but nullable," "optional but non-nullable," and "optional and explicitly nullable" as distinct concepts -- and that's intuitively (+ naively) what I imagined the distinction of `?` on the name vs `| null` in the type accomplishing.

What's the preferred way of expressing the "required but nullable" concept? I can update that here and the few other places that currently have `| null` if that's discouraged.

Reply:
What you have above is required and nullable. In other words, the response *must* include a `finishReason` property, but the value of the property can be `null`.

From an API design perspective, we discourage such use in Azure since it is a) difficult for a nominally typed system to distinguish between the property being present with the value `null` and the property missing (less of a concern for required fields) and b) there is rarely any semantic difference between a missing property and a property with the value `null`, so sending the property with the value `null` only wastes some bytes of network usage.

Reply:
@johanste then would the recommendation be to leave this as required with `| null` and just suppress any relevant linter warnings that may arise?

Reply:
Yes. If the value can actually be `null` on the wire, then that is the appropriate annotation.